python/lsst/meas/deblender/plugins.py

```diff
@@ -0,0 +1,1299 @@
+    (``templateImage`` and ``templateFootprint``) around the peak for all peaks not flagged as
+    ``skip`` or ``deblendedAsPsf``.
+    
+    Parameters
+    ----------
+    debResult: `lsst.meas.deblender.baseline.DeblenderResult`
+        Container for the final deblender results.
+    log: `log.Log`
+        LSST logger for logging purposes.
+    patchEdges: `bool`, optional
+        If True and if the parent Footprint touches pixels with the ``EDGE`` bit set,
+        then grow the parent Footprint to include all symmetric templates.
+
+    Returns
+    -------
+    modified: `bool`
+        If any peaks are not skipped or marked as point sources, ``modified`` is ``True.
+        Otherwise ``modified`` is ``False``.
+    """
+    modified = False
+    # Create the Templates for each peak in each filter
+    for fidx in debResult.filters:
+        dp = debResult.deblendedParents[fidx]
+        imbb = dp.img.getBBox()
+        log.trace('Creating templates for footprint at x0,y0,W,H = %i, %i, %i, %i)', dp.x0, dp.y0, dp.W, dp.H)
+
+        for peaki, pkres in enumerate(dp.peaks):
+            log.trace('Deblending peak %i of %i', peaki, len(dp.peaks))
+            # TODO: Check debResult to see if the peak is deblended as a point source
+            # when comparing all bands, not just a single band
+            if pkres.skip or pkres.deblendedAsPsf:
+                continue
+            modified = True
+            pk = pkres.peak
+            cx, cy = pk.getIx(), pk.getIy()
+            if not imbb.contains(afwGeom.Point2I(cx, cy)):
+                log.trace('Peak center is not inside image; skipping %i', pkres.pki)
+                pkres.setOutOfBounds()
+                continue
+            log.trace('computing template for peak %i at (%i, %i)', pkres.pki, cx, cy)
+            S, patched = butils.buildSymmetricTemplate(dp.maskedImage, dp.fp, pk, dp.avgNoise,
+                                                       True, patchEdges)
+            # SWIG doesn't know how to unpack a std::pair into a 2-tuple...
+            # (in this case, a nested pair)
+            timg, tfoot = S[0], S[1]
+            del S
+
+            if timg is None:
+                log.trace('Peak %i at (%i, %i): failed to build symmetric template', pkres.pki, cx, cy)
+                pkres.setFailedSymmetricTemplate()
+                continue
+
+            if patched:
+                pkres.setPatched()
+
+            # possibly save the original symmetric template
+            if setOrigTemplate:
+                pkres.setOrigTemplate(timg, tfoot)
+            pkres.setTemplate(timg, tfoot)
+    return modified
+
+def rampFluxAtEdge(debResult, log, patchEdges=False):
+    """Adjust flux on the edges of the template footprints.
+    
+    Using the PSF, a peak ``Footprint`` with pixels on the edge of ``footprint``
+    is grown by the ``psffwhm``*1.5 and filled in with zeros.
```

Comment:
You are right. I think I was giving too much information about the process, which assigns the new pixels to 0, but later reassigns them to ramped values to make the template symmetric.